### PR TITLE
feat(hooks): add git-guard PreToolUse hook — block destructive git ops against a dirty tree

### LINKED ISSUE #240: Add git-guard PreToolUse hook: block destructive git ops against a dirty tree
## Why

`git checkout -- <file>` (and `git restore` / `git reset --hard` / `git clean -f`) silently discards uncommitted work. When an agent uses a whole-file revert to undo a single edit — while the file has uncommitted changes that were never staged/committed — the entire change is wiped with no recovery.

This actually happened: a `git checkout -- _transitions.py` used for mutation-testing reset the file to HEAD and erased ~74 lines of uncommitted work. Reconstructed from a captured diff, but it should be impossible to trigger from the agent in the first place.

## What

A new `PreToolUse` Bash guard, `hooks/git-guard.js`, following the same `module.exports = { hooks: [{ matcher, handler }] }` shape as `bash-guard.js` and run via `hook-runner.js`.

Behavior — blocks these only when the **targeted paths are actually dirty**:
- `git checkout -- <path>` / `git checkout .` / `git checkout -f`
- `git restore <path>`
- `git reset --hard`
- `git clean -f`

Design points that keep it low-friction:
- **Path-scoped**: `git checkout -- clean_file` is allowed; `git checkout -- dirty_file` is blocked (`git status --porcelain -- <paths>`). Whole-tree for `reset --hard` / `clean`.
- **Branch switches** (`git checkout main`, `git checkout -b feature`) are never touched.
- The `git status` check only runs when a destructive verb is detected, so normal commands pay nothing.
- Fail-open: not a git repo / git unavailable → never block.

Deny message tells the agent to put a single edit back with Edit (not git), or commit/stash first to keep the work recoverable.

## Wiring

`~/.claude/settings.json` PreToolUse Bash matcher:

```
"command": "node \"$HOME/.claude/hooks/hook-runner.js\" bash-guard.js git-guard.js"
```

## Verification

Tested through the real hook-runner stdin/stdout protocol:
- dirty-file `git checkout -- <file>` → `deny` + message
- clean-file checkout, `git checkout main`, `git checkout -b` → allow
- `rm -rf /` still blocked by `bash-guard.js` (no regression)

## To do in dotfiles

- [ ] Add `hooks/git-guard.js` (currently local-only at `~/.claude/hooks/git-guard.js`)
- [ ] Update the chezmoi-managed `settings.json` Bash hook command to include `git-guard.js`
- [ ] Note the session-start load caveat (takes effect next session; only gates agent tool calls, not the user's own terminal git)

### PR DESCRIPTION
**Pull request by paulnsorensen**
Closes #240.

Blocks destructive git ops that silently discard uncommitted work (`git checkout -- <path>`/`.`/`-f`, `git restore <path>`, `git reset --hard`, `git clean -f`) — but only when the targeted paths actually have uncommitted changes, so a clean tree is never nagged. The classifier handles `sudo`/`env` prefixes, `-C`/`-c` globals, `--` pathspec separation, and `&&`/`||`/`;`/`|`/newline segmentation.

One shared classifier (`agents/lib/git-guard.js`, exporting `shouldBlock`/`denyReason`) drives five per-harness adapters; no detection logic is duplicated:

- **claude + codex** — `PreToolUse` hook (`agents/hooks/git-guard.sh` bridge + registry entry, matcher `Bash`), identical deny schema.
- **cursor** — `beforeShellExecution` hook in cheese-grok (exit 2 = deny).
- **copilot** — `preToolUse` hook under `~/.copilot/hooks` (deny via stdout JSON, double-parsed `toolArgs`).
- **opencode** — plugin `tool.execute.before` on the `bash` tool (`throw` = deny). opencode has no shell hook, but only a plugin can run the dirty-tree check a static `permission.bash` pattern cannot.

Fail-open everywhere (missing lib/node, malformed input, non-repo cwd → allow); opt out with `CLAUDE_GIT_GUARD=0`. 40 bats tests cover every classify branch, the dirty/clean gate, parsing edge cases, each adapter, and deploy wiring. AGENTS.md hook docs updated with the per-harness coverage table.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cursor/plugins/local/cheese-grok/hooks.json [hunk 1]
```diff
@@ -8,6 +8,12 @@
         "matcher": "rm -rf|sudo |chmod 777|git push --force|git reset --hard",
         "failClosed": true,
         "timeout": 2
+      },
+      {
+        "command": "./hooks/git-guard.sh",
+        "type": "command",
+        "matcher": "git\\s+(.*\\s)?(checkout|restore|reset|clean)\\b",
+        "timeout": 5
```
[the remark this answers is not in the document]
Comment (paulnsorensen) [reply cursor/plugins/local/cheese-grok/hooks.json:28]: Fixed — `tests/cursor-plugin.bats` now expects `beforeShellExecution | length == 2` and asserts the appended git-guard entry (its absolute command path + `cheese-grok` _plugin tag). Pushed in 05db565. The CI `test` job was red from this plus three `hook_detect_changes (codex)` baselines and a stray +x bit on `agents/lib/git-guard.js` — all addressed in the same commit.

---
agent on behalf of; paulnsorensen
